source/docs/casper/dapp-dev-guide/writing-contracts/contracts-and-session-code.md

```diff
@@ -0,0 +1,579 @@
+# Smart Contracts and Session Code
```

Comment:
This is excellent information, and well presented in a logical way. 

I'm not sure if it should be split into two pages. There's quite a lot in one place here. On the other hand, it is definitely related information.

Reply:
I think it makes sense to explain the correlation between Session Code and Smart Contracts, otherwise devs don't really know what to do with it and when to use which.